src/Transformer/AbstractEnumTransformer.php

```diff
@@ -0,0 +1,43 @@
+<?php
+declare(strict_types=1);
+
+namespace Diaclone\Transformer;
+
+use Diaclone\Resource\ResourceInterface;
+
+abstract class AbstractEnumTransformer extends AbstractTransformer
+{
+    public function getPropertyValue($data, $property)
+    {
+        if (empty($property)) {
+            return $data->getValue();
+        }
+
+        // convert property to camelCase
+        $getter = 'get' . str_replace(' ', '', ucwords(str_replace(['-', '_'], ' ', $property)));
+        if (method_exists($data, $getter)) {
+            $enum = $data->$getter();
+
+            return $enum->getValue();
```

Comment:
What's the reason you're grabbing the enum value inside here? Why not deal with Enum to Value conversion inside transform?

Reply:
Because inside transform it tries to create the enum class with a constructor, which isn't allowed

Reply:
sorry I'm not following, I thought `transform()` takes exiting raw object (`Enum` class) and transforms it to `string` ($enum->getValue())?

Reply:
I'm an idiot. 

I'll try what you are saying. As I'm looking at this code, something seems wrong about it. I'll refactor it one way or another.